lib/Site.js

```diff
@@ -522,6 +522,11 @@ Site.prototype.generatePages = function () {
   });
   return new Promise((resolve, reject) => {
     Promise.all(processingFiles)
+      .then(() => {
+        this.pages.forEach((page) => {
+          page.collectHeadings(fs.readFileSync(page.resultPath));
```

Comment:
The test fails because `page` is a mock object rather than the an actual `Page` object, so `page.resultPath` is undefined (class attributes do not get mock).

Perhaps redesign the method such that it is unnecessary for Site.js to call `fs.readFileSync(page.resultPath)`?